### DISCUSSION TITLE
UserWarning: Duplicate Operation ID auth_delete for function auth_delete at auth.py   warnings.warn(message, stacklevel=1)

### DISCUSSION THREAD

**Discussion by AlfredThor** [fastapi/fastapi#11240] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
/apps/auth/__init__.py
from fastapi import APIRouter
router = APIRouter()
from apps.auth.auth import *

/apps/auth/auth.py
@router.delete('/del/{id}', summary='delete', operation_id='auth_delete')
async def auth_delete(id: int):
    print(id)
    return {'code': 200, 'message': 'OK!'}

main.py
from apps import auth
app.add_middleware(
    CORSMiddleware,
    allow_origins=['*'],
    allow_methods=['*'],
    allow_headers=['*'],
    allow_credentials=True,
)
app.include_router(auth.router, prefix='/auth', tags=['用户'])
```

### Description

When I specify operation_id or do not specify operation_id in the router decorator, a UserWaring warning appears

### Operating System

macOS

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.9

### Additional Context

/Users/Alfred/venv/AppAgora/lib/python3.9/site-packages/fastapi/openapi/utils.py:207: UserWarning: Duplicate Operation ID auth_upd_auth_update__auth_id__put for function auth_upd at /Users/Alfred/PycharmProjects/AppAgora/apps/auth/auth.py
  warnings.warn(message, stacklevel=1)

/Users/Alfred/venv/AppAgora/lib/python3.9/site-packages/fastapi/openapi/utils.py:207: UserWarning: Duplicate Operation ID auth_delete for function auth_delete at /Users/Alfred/PycharmProjects/AppAgora/apps/auth/auth.py
  warnings.warn(message, stacklevel=1)

The upper part is a warning message with no operation_id set, and the lower part is a warning message with operation_id set

**Comment by JavierSanchezCastro** [comment]
Based on your output, it seems like you're using the same operation_id somewhere else, in another function.

**Reply by AlfredThor** [reply]
I looked at the source code and also consulted with ChatGPT. The action ID in this alarm is mainly used to identify different actions in the OpenAPI documentation, so it should not cause problems for the program. I checked all the action ids used to make sure each one was unique, but the warning message still appeared.

**Comment by Yuerchu** [comment, marked as answer]
I encountered the same problem today. Here's my solution. I don't know if it will be useful to you?

```python
# ./routers/admin.py
from fastapi import APIRouter

Router = APIRouter(prefix="/admin", tags=["admin"])

@Router.get("/")
def read_root():
    return {"Hello": "Admin"}

# ./app.py
from fastapi import FastAPI
from routers import admin

app = FastAPI()

def mount_backend():
    app.include_router(admin.Router)

@app.get("/")
def read_root():
    return {"Hello": "World"}

# ./main.py
from app import app, mount_backend
import uvicorn

mount_backend()

if __name__ == "__main__":
    # uvicorn.run("main:app")
    uvicorn.run("app:app")
```

**Reply by YuriiMotov** [reply]
Just to clarify what is happening when you use `uvicorn.run("main:app")`:

it imports `main.py` module using `importlib.import_module()` and this causes the execution of top level code of that module.
So, top-level code is executed twice: first when you run `python main.py`, second when `uvicorn.run()` imports `main.py`.

Another way to solve this is to pass the instance of FastAPI application (`app`) instead of the string "main:app":

```
import uvicorn
from app import app, mount_backend

mount_backend()

if __name__ == "__main__":
    uvicorn.run(app)
```
